sdk/containerregistry/azure-containerregistry/azure/containerregistry/_authentication_policy.py

```diff
@@ -22,11 +22,13 @@ class ContainerRegistryChallengePolicy(HTTPPolicy):
     """Authentication policy for ACR which accepts a challenge"""
 
     def __init__(self, credential, endpoint, **kwargs):
-        # type: (TokenCredential, str, **Any) -> None
+        # type: (Optional[TokenCredential], str, **Any) -> None
```

Comment:
Is this constructor public?

Reply:
It is public. The hint is not correct, the credential could be None, we'll create `AnonymousACRExchangeClient` in this case.